api/types/registry/registry.go

```diff
@@ -102,3 +105,15 @@ type SearchResults struct {
 	// Results is a slice containing the actual results for the search
 	Results []SearchResult `json:"results"`
 }
+
+// DistributionInspect describes the result obtained from contacting the
+// registry to retrieve image metadata
+type DistributionInspect struct {
```

Comment:
This data structure is making the assumption that a name may map to single a image id, which is untrue. For manifest lists, the name may map to multiple image ids (which are effectively the config digest).